eos/utils/expression-visitors.cc

```diff
@@ -385,20 +386,46 @@ namespace eos::exp
     {
         std::set<std::string> kinematic_set;
 
-        const auto & kinematics_values = e.kinematics_specification.values;
-        const auto & kinematics_aliases = e.kinematics_specification.aliases;
+        const auto & entries = impl::observable_entries;
+        const auto & it = entries.find(e.observable_name);
 
-        // Set or alias kinematic specifications
-        for (const auto & value : kinematics_values)
+        // check if 'e' matches the name of a known observable
+        if (it != entries.end())
         {
-            kinematic_set.insert(value.first);
+            const auto entry = it->second;
+
+            kinematic_set.insert(entry->begin_kinematic_variables(), entry->end_kinematic_variables());
+
+            const auto & kinematics_values = e.kinematics_specification.values;
+            const auto & kinematics_aliases = e.kinematics_specification.aliases;
+
+            // Set or alias kinematic specifications
+            for (const auto & value : kinematics_values)
+            {
+                kinematic_set.insert(value.first);
+            }
+            for (const auto & alias : kinematics_aliases)
+            {
+                kinematic_set.insert(alias.second);
+            }
+
+            return kinematic_set;
         }
-        for (const auto & alias : kinematics_aliases)
+
+        // otherwise check if 'e' matches the name of a parameter
+        if (kinematic_set.empty())
```

Comment:
I think the "is it a parameter" check should be put into an ``else`` block. Only look for this being a parameter if there is no observable entry. This is how ``Observable::make`` treats it.

Reply:
It is already the case since the previous `if` ends with a `return` statement. But I will make it explicit